cleverpush/src/main/java/com/cleverpush/stories/StoryView.java

```diff
@@ -0,0 +1,140 @@
+                SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+                for (int i = 0; i < stories.size(); i++) {
+                    if (sharedPreferences.getString(CleverPushPreferences.APP_OPENED_STORIES, "").contains(stories.get(i).getId())) {
+                        stories.get(i).setOpened(true);
+                    } else {
+                        stories.get(i).setOpened(false);
+                    }
+                }
+                ((Activity) context).runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        displayStoryHead(stories);
+                    }
+                });
+            }
+
+            @Override
+            public void onFailure(int statusCode, String response, Throwable throwable) {
+                loading = false;
+                Log.e(TAG, "Something went wrong when loading stories." +
+                        "\nStatus code: " + statusCode +
+                        "\nResponse: " + response
+                );
+            }
+        };
+    }
+
+    private void displayStoryHead(ArrayList<Story> stories) {
+        View root = LayoutInflater.from(context).inflate(R.layout.story_view, this, true);
+
+        RelativeLayout rlMain = root.findViewById(R.id.rlMain);
+        rlMain.setBackgroundColor(attrArray.getColor(R.styleable.StoryView_background_color, DEFAULT_BACKGROUND_COLOR));
+        ViewGroup.LayoutParams params = rlMain.getLayoutParams();
+        params.height = (int) attrArray.getDimension(R.styleable.StoryView_story_view_height, ViewGroup.LayoutParams.WRAP_CONTENT);
+        params.width = (int) attrArray.getDimension(R.styleable.StoryView_story_view_width, ViewGroup.LayoutParams.WRAP_CONTENT);
+        rlMain.setLayoutParams(params);
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager((Activity) context, LinearLayoutManager.HORIZONTAL, false);
+        RecyclerView recyclerView = root.findViewById(R.id.rvStories);
+        storyViewListAdapter = new StoryViewListAdapter((Activity) context, stories, attrArray, getOnItemClickListener(stories, recyclerView));
+        recyclerView.setLayoutManager(linearLayoutManager);
+        recyclerView.setAdapter(storyViewListAdapter);
+    }
+
+    private OnItemClickListener getOnItemClickListener(ArrayList<Story> stories, RecyclerView recyclerView) {
+        return new OnItemClickListener() {
+            @Override
+            public void onClicked(int position) {
+                SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+                SharedPreferences.Editor editor = sharedPreferences.edit();
+                if (sharedPreferences.getString(CleverPushPreferences.APP_OPENED_STORIES, "").isEmpty()) {
+                    editor.putString(CleverPushPreferences.APP_OPENED_STORIES, stories.get(position).getId()).apply();
```

Comment:
you can export stories.get(position).getId() as variable and re-use it inside the else statement

same with sharedPreferences.getString(CleverPushPreferences.APP_OPENED_STORIES, "")